### ISSUE TITLE
source account creation data for login.gov and ID.me

### ISSUE THREAD

**Issue by djmunoz1** [department-of-veterans-affairs/va.gov-team#63085]
As an admin I want to be able to track users who create an account after receiving the GovDelivery email so I know how effective each campaign is. 

Currently, we have a view of the numbers of users who create accounts by each set of users we last emailed. We have analytics from GovDelivery on click-through rates. We have logs of user ICNs and can associate to account credential type. We would like to be able to associate account creation to a "source," so anyone who receives a GovDelivery email, then clicks the link within and subsequently creates an account, we can track.

ACCEPTANCE:
- discovery of VA.gov repo and data logs to understand availability of account creation data
- consult Core team for VA.gov data availability via DataDog (for LGN and ID.me)
- consult login.gov for account creation data

DEF OF DONE:
- Track source of account creation so we can see who created an account after receiving an email

Questions:
- Will GovDelivery will send back the emails and the actions from each individual? Specifically, which users clicked through?
- Can we filter out what the traffic does - users coming from our email campaigns (source)? Need to know the specific page each email is going to send them to

**Comment by sdickson-gt** [comment]
Have asked Travis and Joe in the identity-engineers slack channel:
For Login.gov adoption, we’d like to be able to attribute Login.gov and ID.me account logins and CSP account creations with some of the upcoming planned comms. For the CSP account creations, can you tell me how to tell for a user who just logged in, whether its a new or existing account?  Is the account creation tracked in the DataDog logs, for example, I’ve been searching for the following to indicate successful authentication, but this doesn't indicate new vs existing CSP accounts AFAIK. cc 
@Lauren Morris
SSOe: @message_content:"SSO: LOGIN"
SiS: @message_content:*SignInController*callback

**Comment by sdickson-gt** [comment]
Following datadog query will show when a user is new VA.gov user: "login userverifier new".  These users may already be in MPI, but these users are new to VA.gov.

**Comment by sdickson-gt** [comment]
Followed up with Joe: To help us report on outreach to our reactivation and organic user population, could these logs be enhanced with icn? Would icn exist for all users if they create an ID-verified account, regardless of whether the user was previously known or not?

**Comment by sdickson-gt** [comment]
The GD discovery is ongoing - haven't heard back yet.  Will follow up with Joe/Trevor regarding enhancing the logs.

**Comment by sdickson-gt** [comment]
GD provided a means to customize headers, so we should now be able to report how many users click through using automated integration from DataDog to Domo.  TBD if datadog is using some form of correlationID so we can correlate the entry points (anonymous user) with subsequent login success / new credential events. 

For the events "new user credential to VA.gov", Joe has created a ticket to enhance the logs with icn: https://github.com/orgs/department-of-veterans-affairs/projects/1137/views/12?pane=issue&itemId=50548481

**Comment by sdickson-gt** [comment]
We have a couple different methods to track who from an email campaign introduced a new account to the VA. Both of these will require some development
- VA.gov logs / DataDog (issue above is complete)
- IAM SSOe logs / Splunk
Again, we can assume that most new accounts to the VA were also newly created at Login.gov, but without getting data from Login.gov we don't know that for certain.  Some users may have had a Login.gov account, but hadn't used it with VA yet.

Still need to investigate traceablity of email click through to authentication.  Couple options
- DataDog - looking into whether there is a correlating identifier to correlate the entry point (with the GovDelivery parameters) to an authentication log
- GA - GA can give a general idea of what users are doing after coming to VA after clicking the link. Caveat is that it is just a sampling of all users.  Need to investigate whether GA Behavior Flow can be configured with a specific starting page (it appears to only list the top 20 entry pages)

**Comment by sdickson-gt** [comment]
DD query users arriving from GD link: https://vagov.ddog-gov.com/logs?query=%40http.url_details.queryString.utm_content%3A011924testemail%20&cols=%40remote_ip.geoip.country.name%2C%40payload.user_uuid%2C%40named_tags.remote_ip%2C%40payload.sso_cookie_contents.patientIcn&index=%2A&messageDisplay=inline&refresh_mode=sliding&saved-view-id=1170&storage=hot&stream_sort=desc&view=spans&viz=stream&from_ts=1706203929515&to_ts=1706207529515&live=true

Still TBD whether these can be correlated to a subsequent authentication event.   Not clear how some include the params next=loginModal&postLogin=true.  I'm not able to replicate this - not seeing these after authenticating.
 utm_content=011924testemail&utm_medium=email&utm_name=SSIT&utm_source=govdelivery&next=loginModal&postLogin=true

**Comment by sdickson-gt** [comment]
I wasn't able to find any correlating data between the initial click-through event log and the post-authentication logs.  I updated DD to create metrics associated with GD click throughs and attributes to distinguish between the message campaigns.  I created a connector to pull this from DataDog and updated the dashboard with the visualization.  https://va-gov.domo.com/page/923236635?userId=1480079782
